api/server/middleware/debug.go

```diff
@@ -42,11 +61,15 @@ func DebugRequestMiddleware(handler func(ctx context.Context, w http.ResponseWri
 		var postForm map[string]interface{}
 		if err := json.Unmarshal(b, &postForm); err == nil {
 			maskSecretKeys(postForm)
-			formStr, errMarshal := json.Marshal(postForm)
-			if errMarshal == nil {
-				log.G(ctx).Debugf("form data: %s", string(formStr))
+			// TODO(thaJeztah): is there a better way to detect if we're using JSON-formatted logs?
+			if _, ok := logger.Logger.Formatter.(*logrus.JSONFormatter); ok {
+				fields["form-data"] = postForm
 			} else {
-				log.G(ctx).Debugf("form data: %q", postForm)
+				if data, err := json.Marshal(postForm); err != nil {
+					fields["form-data"] = postForm
+				} else {
+					fields["form-data"] = string(data)
+				}
```

Comment:
What do we lose if we just use `fields["form-data"] = postForm` without bothering to marshal the JSON ourselves for non-JSON formatters?

Reply:
The output would be in Go's formatting used for `map[string]any`. It's still readable, but slightly less useful if you want to (e.g.) replicate a request with `cURL`;

```
form-data="map[Driver:local Name:foo]"
```